lisa/util/shell.py

```diff
@@ -499,6 +527,83 @@ def _purepath_to_str(
             path = str(path)
         return path
 
+    def _establish_jump_boxes(self, address: str, port: int) -> Any:
+        jump_boxes_runbook = development.get_jump_boxes()
+        sock: Any = None
+        is_trace_enabled = development.is_trace_enabled()
+        if is_trace_enabled:
+            jb_logger = _get_jump_box_logger()
+            jb_logger.debug(f"proxy sock: {sock}")
+
+        for index, runbook in enumerate(jump_boxes_runbook):
+            if is_trace_enabled:
+                jb_logger.debug(f"creating connection from source: {runbook} ")
+            client = paramiko.SSHClient()
+            client.set_missing_host_key_policy(paramiko.MissingHostKeyPolicy())
+            client.connect(
+                hostname=runbook.address,
+                port=runbook.port,
+                username=runbook.username,
+                password=runbook.password,
+                key_filename=runbook.private_key_file,
```

Comment:
Does all jumpbox need to have same key?

Reply:
It doesn't need. The key can be configured for each jumpbox like address or port.